src/main/java/io/github/thebusybiscuit/slimefun4/integrations/OrebfuscatorIntegration.java

```diff
@@ -0,0 +1,62 @@
+package io.github.thebusybiscuit.slimefun4.integrations;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+
+import org.bukkit.Bukkit;
+import org.bukkit.block.Block;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+import io.github.thebusybiscuit.slimefun4.api.events.BlockPlacerPlaceEvent;
+import io.github.thebusybiscuit.slimefun4.api.events.ExplosiveToolBreakBlocksEvent;
+import io.github.thebusybiscuit.slimefun4.api.events.ReactorExplodeEvent;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import net.imprex.orebfuscator.api.OrebfuscatorService;
+
+/**
+ * 
+ * This handles block breaks with orebfuscator
+ * 
+ * @author NgLoader
+ *
+ */
+public class OrebfuscatorIntegration implements Listener {
+
+	private final SlimefunPlugin plugin;
+	private OrebfuscatorService service;
+
+	public OrebfuscatorIntegration(@Nonnull SlimefunPlugin plugin) {
+        this.plugin = plugin;
+    }
+
+	/**
+	 * Init orebfuscation service and register listener
+	 */
+    public void register() {
+        this.service = Bukkit.getServer().getServicesManager().getRegistration(OrebfuscatorService.class).getProvider();
+
+        Bukkit.getServer().getPluginManager().registerEvents(this, this.plugin);
+    }
+
+    @EventHandler
+    public void onBlockPlacerPlace(BlockPlacerPlaceEvent event) {
+        this.service.deobfuscate(Arrays.asList(event.getBlock(), event.getBlockPlacer()));
+    }
+
+    @EventHandler
+    public void onExplosiveToolBreakBlocks(ExplosiveToolBreakBlocksEvent event) {
+        Set<Block> blocks = new HashSet<>();
+        blocks.addAll(event.getAdditionalBlocks());
+        blocks.add(event.getPrimaryBlock());
+        this.service.deobfuscate(blocks);
+    }
+
+    @EventHandler
+    public void onReactorExplode(ReactorExplodeEvent event) {
+        this.service.deobfuscate(Arrays.asList(event.getLocation().getBlock()));
+    }
```

Comment:
One thing I forgot:

None of these Event Handlers affect the event's outcome, so they should all listen on the priority `MONITOR` and have `ignoreCancelled` set to `true`.

This helps to reduce unnecessary deobfuscations if the event did not actually process.

Reply:
Oh, okay

Reply:
But then I need to set ``ignoreCancelled`` to ``false``

Reply:
`ignoreCancelled` means that the `EventHandler` will ignore cancelled events.
That means that your code won't be triggered if the event was cancelled. So `ignoreCancelled=true` is what you want.

Example:
If the BlockPlacer event gets cancelled, no block is placed. So we don't need to trigger an Orebfuscator update either since no change was made to the world.

Here is a javadocs ref:
https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/EventHandler.html

Reply:
Oh xD
But the other line says ``Define if the handler ignores a cancelled event.`` so I was thinking when the value is ``true`` the event will be canceled